## 민간 아파트 분양가격 동향 2023-08-21
## (데이터 전처리과정)

In [105]:
import pandas as pd
df = pd.read_csv('https://bit.ly/Private-apartment-prices', encoding='cp949')
df

지역명                     규모구분    연도   월 분양가격(제곱미터)
0     서울                     모든면적  2015  10       5841
1     서울            전용면적 60제곱미터이하  2015  10       5652
2     서울   전용면적 60제곱미터초과 85제곱미터이하  2015  10       5882
3     서울  전용면적 85제곱미터초과 102제곱미터이하  2015  10       5721
4     서울           전용면적 102제곱미터초과  2015  10       5879
...   ..                      ...   ...  ..        ...
7900  제주                     모든면적  2023   6       7326
7901  제주            전용면적 60제곱미터이하  2023   6       7381
7902  제주   전용면적 60제곱미터초과 85제곱미터이하  2023   6       7084
7903  제주  전용면적 85제곱미터초과 102제곱미터이하  2023   6       6639
7904  제주           전용면적 102제곱미터초과  2023   6       7506

[7905 rows x 5 columns]

In [106]:
#데이터 프레임 정보확인
df.info()
#'분양 가격'에서 7358개의 non-null 확인. -> 547개 결측치 처리

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7905 entries, 0 to 7904
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   지역명         7905 non-null   object
 1   규모구분        7905 non-null   object
 2   연도          7905 non-null   int64 
 3   월           7905 non-null   int64 
 4   분양가격(제곱미터)  7358 non-null   object
dtypes: int64(2), object(3)
memory usage: 308.9+ KB


## 1. 컬럼명 수정하기
'분양가격(제곱미터) -> '분양가격'

In [107]:
df.rename(columns={'분양가격(제곱미터)':'분양가격'}, inplace=True)

In [108]:
df

지역명                     규모구분    연도   월  분양가격
0     서울                     모든면적  2015  10  5841
1     서울            전용면적 60제곱미터이하  2015  10  5652
2     서울   전용면적 60제곱미터초과 85제곱미터이하  2015  10  5882
3     서울  전용면적 85제곱미터초과 102제곱미터이하  2015  10  5721
4     서울           전용면적 102제곱미터초과  2015  10  5879
...   ..                      ...   ...  ..   ...
7900  제주                     모든면적  2023   6  7326
7901  제주            전용면적 60제곱미터이하  2023   6  7381
7902  제주   전용면적 60제곱미터초과 85제곱미터이하  2023   6  7084
7903  제주  전용면적 85제곱미터초과 102제곱미터이하  2023   6  6639
7904  제주           전용면적 102제곱미터초과  2023   6  7506

[7905 rows x 5 columns]

### 2. "분양가격(m^2)"에 있는 데이터 공백을 삭제하기
#str.strip() 이용

In [109]:
import pandas as pd

has_space = df['분양가격'].str.contains(' ', regex=False)
print(has_space.sum()) #True가 36개 있는 것을 확인

36


In [110]:
df['분양가격'] = df['분양가격'].str.strip()

In [111]:
has_space = df['분양가격'].str.contains(' ', regex=False)
print(has_space.sum()) #공백이 모두 지워진 것을 확인

0


### 3. "분양가격"
1. 데이터 안에 있는 ","를 삭제하기
2. 데이터 안에 있는 "-"를 삭제하기
3. 데이터가 없는경우("") 0으로 대체하기

In [112]:
has_comma = df['분양가격'].str.contains(',', regex=False)
has_dash = df['분양가격'].str.contains('-', regex=False)
has_none = df['분양가격'].isna()
print(has_comma.sum()) # 데이터 안에 ,가 없음
print(has_dash.sum()) # 데이터 안에 - 가 없음
print(has_none.sum()) # 데이터가 없는경우 547개

0
0
547


In [113]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7905 entries, 0 to 7904
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   지역명     7905 non-null   object
 1   규모구분    7905 non-null   object
 2   연도      7905 non-null   int64 
 3   월       7905 non-null   int64 
 4   분양가격    7358 non-null   object
dtypes: int64(2), object(3)
memory usage: 308.9+ KB


In [114]:
df['분양가격'].head(5)

0    5841
1    5652
2    5882
3    5721
4    5879
Name: 분양가격, dtype: object

In [115]:
(df['분양가격'] == "").sum()  # ""데이터가 총 36개

36

In [116]:
df['분양가격'] = df['분양가격'].str.replace(',','')
df['분양가격'] = df['분양가격'].str.replace('-','')
df.loc[df['분양가격']== "", "분양가격"] = "0"

In [117]:
(df['분양가격'] == "0").sum()  # "0"데이터가 총 36개

36

### 4. "분양가격"의 Nan 데이터를 삭제하기

In [118]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7905 entries, 0 to 7904
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   지역명     7905 non-null   object
 1   규모구분    7905 non-null   object
 2   연도      7905 non-null   int64 
 3   월       7905 non-null   int64 
 4   분양가격    7358 non-null   object
dtypes: int64(2), object(3)
memory usage: 308.9+ KB


In [119]:
# Nan 데이터를 삭제
df.dropna(subset=['분양가격'], inplace=True)

In [120]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7358 entries, 0 to 7904
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   지역명     7358 non-null   object
 1   규모구분    7358 non-null   object
 2   연도      7358 non-null   int64 
 3   월       7358 non-null   int64 
 4   분양가격    7358 non-null   object
dtypes: int64(2), object(3)
memory usage: 344.9+ KB


### 5. "분양가격"의 dtype를 int로 변환하기

In [121]:
df['분양가격'] = df['분양가격'].astype('int')

In [122]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7358 entries, 0 to 7904
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   지역명     7358 non-null   object
 1   규모구분    7358 non-null   object
 2   연도      7358 non-null   int64 
 3   월       7358 non-null   int64 
 4   분양가격    7358 non-null   int64 
dtypes: int64(3), object(2)
memory usage: 344.9+ KB


### 6. "규모구분" :  데이터 내의 "전용면적" 글자를 삭제해주세요

In [124]:
df['규모구분'].str.contains("전용면적").sum() #5792개의 행에 "전용면적"이 포함되어있음

5792

In [125]:
df['규모구분'] = df['규모구분'].str.replace('전용면적',"")

In [126]:
df['규모구분'].str.contains("전용면적").sum() # 지워진 것을 확인

0

### 7. "규모구분" 컬럼을 기준으로 value count를 해주세요

In [128]:
print(df['규모구분'].value_counts())

모든면적                   1566
 60제곱미터초과 85제곱미터이하     1563
 60제곱미터이하              1507
 102제곱미터초과             1438
 85제곱미터초과 102제곱미터이하    1284
Name: 규모구분, dtype: int64


### 8. 지역명별로 평균 분양가격을 확인

In [132]:
df.groupby('지역명')['분양가격'].mean()

지역명
강원    2817.401766
경기    4412.705376
경남    3153.317982
경북    2941.656885
광주    3557.351899
대구    4213.241228
대전    3378.236620
부산    4156.941685
서울    7936.511983
세종    3217.769821
울산    3963.243323
인천    4180.193059
전남    2627.616740
전북    2571.648590
제주    4410.950000
충남    2819.103139
충북    2542.922246
Name: 분양가격, dtype: float64

### 9. 지역별 최고 비싼 분양가를 확인
1. 가장 비싼 분양가를 나타내는 지역은 어디이며 얼마 인가요? - 
2. 제주의 최고가는? - 

In [135]:
#1
df.sort_values(by='분양가격', ascending = False).head(1)

지역명                 규모구분    연도  월   분양가격
4593  서울   85제곱미터초과 102제곱미터이하  2020  4  13835

> 가장 비싼 분양가를 나타내는 지역은 서울이며, 제곱미터 당 13835원이다.

In [139]:
#2
df[df['지역명']=='제주']['분양가격'].max()

8410

> 제주에서 최고가는 제곱미터당 8410원이다.

### 작업파일을 저장합니다.(.csv)

In [140]:
df.to_csv('Private-apartment-prices_clean.csv', index=False)